src/main/kotlin/no/nav/familie/oppdrag/tss/TssOppslagService.kt

```diff
@@ -3,29 +3,36 @@ package no.nav.familie.oppdrag.tss
 import no.nav.familie.kontrakter.ba.tss.SamhandlerAdresse
 import no.nav.familie.kontrakter.ba.tss.SamhandlerInfo
 import no.nav.familie.kontrakter.ba.tss.SøkSamhandlerInfo
+import no.nav.familie.oppdrag.tss.TssSamhandlerIdentType.ORGNR
 import no.rtv.namespacetss.SamhAvdPraType
 import no.rtv.namespacetss.Samhandler
 import no.rtv.namespacetss.TssSamhandlerData
 import no.rtv.namespacetss.TypeKomp940
 import no.rtv.namespacetss.TypeOD910
 import no.rtv.namespacetss.TypeSamhAdr
 import no.rtv.namespacetss.TypeSamhAvd
+import org.slf4j.LoggerFactory
 import org.springframework.stereotype.Service
 
 @Service
 class TssOppslagService(private val tssMQClient: TssMQClient) {
+    private val logger = LoggerFactory.getLogger(TssOppslagService::class.java)
 
-    fun hentSamhandlerDataForOrganisasjonB910(orgNr: String): TypeOD910 {
-        val samhandlerData = tssMQClient.getOrgInfo(orgNr)
-        validateB910response(orgNr, samhandlerData)
+    fun hentSamhandlerDataForOrganisasjonB910(tssSamhandlerIdent: TssSamhandlerIdent): TypeOD910 {
+        val samhandlerData = tssMQClient.getOrgInfo(tssSamhandlerIdent)
+        validateB910response(tssSamhandlerIdent, samhandlerData)
         return samhandlerData.tssOutputData.samhandlerODataB910
     }
 
-    fun hentSamhandlerDataForOrganisasjon(orgNr: String): SamhandlerInfo {
-        val samhandlerData910 = hentSamhandlerDataForOrganisasjonB910(orgNr)
+    fun hentSamhandlerInformasjon(tssSamhandlerIdent: TssSamhandlerIdent): SamhandlerInfo {
+        val samhandlerData910 = hentSamhandlerDataForOrganisasjonB910(tssSamhandlerIdent)
 
+        if (samhandlerData910.antallSamhandlere.toInt() > 1) {
+            logger.warn("Fant flere samhandlere for $tssSamhandlerIdent")
+        }
         val enkeltSamhandler = samhandlerData910.enkeltSamhandler.first()
-        return mapSamhandler(enkeltSamhandler)
+        val orgNr = if (tssSamhandlerIdent.type == ORGNR) tssSamhandlerIdent.ident else null
+        return mapSamhandler(enkeltSamhandler, if (tssSamhandlerIdent.type == ORGNR) tssSamhandlerIdent.ident else null)
```

Comment:
Her kan vel `orgNr` brukes som parameter i funksjonen `mapSamhandler`?
```suggestion
        return mapSamhandler(enkeltSamhandler, orgNr)
```